crypto/dh/dh_lib.c

```diff
@@ -246,14 +246,21 @@ void DH_get0_pqg(const DH *dh, BIGNUM **p, BIGNUM **q, BIGNUM **g)
 int DH_set0_pqg(DH *dh, BIGNUM *p, BIGNUM *q, BIGNUM *g)
 {
     /* q is optional */
-    if (p == NULL || g == NULL)
+    if ((p == NULL && dh->p == NULL)
+        || (g == NULL && dh->g == NULL))
```

Comment:
I continue to see it being used in the way it was originally designed, and as they currently operate. The only thing we are (potentially) doing here is adding some extra fault tolerance.